app/actions/remote/custom_emoji.ts

```diff
@@ -66,3 +67,39 @@ export const searchCustomEmojis = async (serverUrl: string, term: string) => {
         return {error};
     }
 };
+
+const names = new Set<string>();
+const debouncedFetchEmojiByNames = debounce(async (serverUrl: string) => {
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    const promises: Array<Promise<CustomEmoji>> = [];
+    for (const name of names) {
+        promises.push(client.getCustomEmojiByName(name));
+    }
```

Comment:
This seems horrible, but I've seen there is no other way with the current API (unless we fetch all custom emojis on startup, that in some circunstances may not be viable due to the amount some instances have).

Wonder if we can leverage graphQL for anything related to this.

Reply:
yes, the current API don't allow us to do much, that is why I'm sort of batching the requests here.. I would say that we could leverage gQL by adding a graph that will allow us to fetch a list of emojis by name etc..

There are systems that have 4K+ emojis and fetching them all is basically something that we should avoid at all cost.